code/modules/projectiles/gun.dm

```diff
@@ -422,14 +422,26 @@
 		reskin_gun(user)
 
 /obj/item/gun/proc/reskin_gun(mob/M)
-	var/choice = input(M,"Warning, you can only reskin your weapon once!","Reskin Gun") in options
+	var/list/skins = list()
+	for(var/I in options)
+		var/image/gun_image = image(icon, icon_state = options[I])
+		skins[I] = gun_image
+	var/choice = show_radial_menu(M, src, skins, null, 40, CALLBACK(src, PROC_REF(radial_check), M), TRUE)
```

Comment:
Can you use keyword parameters here instead of using a null parameter?
```suggestion
	var/choice = show_radial_menu(M, src, skins, radius = 40, custom_check = CALLBACK(src, PROC_REF(radial_check), M), require_near = TRUE)
```